### PR TITLE
Pin dependencies

### PR DESCRIPTION
**Pull request by Kern--**
Signed-off-by: Kern Walster <walster@amazon.com>

*Issue #, if available:*
Closes https://github.com/firecracker-microvm/firecracker-go-sdk/issues/446

*Description of changes:*

This pins our binary dependencies (CNI + project checks) in a way that is compatible with go versions >= 1.14.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/cmd/snapshotting/Makefile [hunk 1]
```diff
@@ -18,8 +18,15 @@ ARCH=$(shell uname -m)
 GID = $(shell id -g)
 
 PWD=$(shell pwd)
+GOBIN=$(PWD)/bin
 FC_TEST_DATA_PATH?=$(PWD)
 
+define install_go
+	cd .hack \
+		GO111MODULE=on GOBIN=$(GOBIN) go get $(1) && \
+		GO111MODULE=on GOBIN=$(GOBIN) go install $(1)
```
Comment (austinvazquez) [inline examples/cmd/snapshotting/Makefile:27]: Do we need to do an `|| true` here since one will fail?

Comment (Kern--) [reply examples/cmd/snapshotting/Makefile:27]: No, it seems like what we actually need is this (ignoring setting env vars for clarity):

go 1.14/1.15:
```
go get $PACKAGE@$VERSION
go install $PACKAGE
```
go 1.16+
```
go install $PACKAGE@$VERSION
```

I've been banging my head against a wall trying to figure out a reasonable way to do this either in the Makefile or in portable shell scripting, but maybe it's worth just having the 10 or so gross if/else make rules.

**Review by austinvazquez — APPROVED** [review]
+1,  this looks great all things considered and let's us keep older versions for users.

**Comment by austinvazquez** [comment]
Looks like the buildkite pipeline installs a plugin as well. We can probably replace that with an explicit make target call to install or let make dependencies handle it.
Reference: https://github.com/firecracker-microvm/firecracker-go-sdk/blob/da8e44af507a3cc6e6dab14c5e4a13d9cf539f38/.buildkite/pipeline.yml#L80

**Comment by austinvazquez** [comment]
The load snapshot integration tests will intermittently fail which #418 is trying to address. That is outside the scope of this change which is why I reran for the green.

**Comment by Kern--** [comment]
@Joffref This will take the place of your PR. Can you please take a look?

**Review by ginglis13 — APPROVED** [review]

**Comment by Joffref** [comment]
@Kern-- LGTM. 
Also, you can mention that it closes issue #446.

**Review by kzys — APPROVED** [review]

**Pull request merged**